toolchain/check/eval.cpp

```diff
@@ -177,6 +185,27 @@ static auto GetConstantValue(Context& context, SemIR::TypeBlockId type_block_id,
   return type_block_id;
 }
 
+// The constant value of a generic instance is the generic instance with the
+// corresponding constant values for its arguments.
+static auto GetConstantValue(Context& context,
+                             SemIR::GenericInstanceId instance_id, Phase* phase)
+    -> SemIR::GenericInstanceId {
+  if (!instance_id.is_valid()) {
+    return SemIR::GenericInstanceId::Invalid;
+  }
+
+  auto& instance = context.generic_instances().Get(instance_id);
```

Comment:
```suggestion
  const auto& instance = context.generic_instances().Get(instance_id);
```

Minor, just wary of the non-const reference.